src/main/webapp/dev/js/main/instructorCourseEnrollPage.js

```diff
@@ -43,39 +69,135 @@ const handsontable = new Handsontable(container, {
+    const toggleChevron = $panelHeading.parent().find('.glyphicon-chevron-down, .glyphicon-chevron-up');
+
+    // perform AJAX only if existing students' spreadsheet is empty
+    if (getSpreadsheetLength(dataHandsontable.getData()) === 0) {
+        getAjaxStudentList(displayIcon)
+                .then((data) => {
+                    if (data.students.length === 0) {
+                        displayNoExistingStudents(displayIcon);
+                    } else {
+                        loadExistingStudentsData(data.students);
+                        adjustStudentsPanelView($panelHeading, panelCollapse,
+                                displayIcon, toggleChevron);
+                    }
+                }).catch(() => {
+                    displayErrorExecutingAjax(displayIcon);
+                });
+    } else {
+        adjustStudentsPanelView($panelHeading, panelCollapse,
+                displayIcon, toggleChevron);
+    }
+}
+
+/**
+ * Expands "New students" panel. Spreadsheet interface would be shown after expansion, including its affiliated buttons.
+ * The panel will be collapsed otherwise if the spreadsheet interface is already shown.
+ */
+function expandCollapseNewStudentsPanel() {
+    const $panelHeading = $(this);
+    const panelCollapse = $panelHeading.parent().children('.panel-collapse');
+    const displayIcon = $panelHeading.children('.display-icon');
+    const toggleChevron = $panelHeading.parent().find('.glyphicon-chevron-down, .glyphicon-chevron-up');
+
+    toggleStudentsPanel($panelHeading, panelCollapse, displayIcon, toggleChevron);
+    enrollHandsontable.render();
+
+    if ($(panelCollapse[0]).hasClass('checked')) { // if panel is shown
+        $('.enroll-students-spreadsheet-buttons').show();
+    } else {
+        $('.enroll-students-spreadsheet-buttons').hide();
+    }
+}
+
 $(document).ready(() => {
     prepareInstructorPages();
+    $('#enroll-spreadsheet').on('click', expandCollapseNewStudentsPanel);
+    $('#enroll-spreadsheet').trigger('click');
+
+    $('#existing-data-spreadsheet').click(expandCollapseExistingStudentsPanel);
 
     if ($('#enrollstudents').val()) {
         const allData = $('#enrollstudents').val().split('\n'); // data in the table including column headers (string format)
 
         const columnHeaders = allData[0].split('|');
-        handsontable.updateSettings({
+        enrollHandsontable.updateSettings({
             colHeaders: columnHeaders,
         });
 
         const spreadsheetDataRows = allData.slice(1);
         if (spreadsheetDataRows.length > 0) {
             const data = getUpdatedData(spreadsheetDataRows);
-            handsontable.loadData(data); // Reset all cells in the grid to contain data from the data array
+            enrollHandsontable.loadData(data); // Reset all cells in the grid to contain data from the data array
         }
     }
 
-    $('#addEmptyRows').click(() => {
+    $('#button_add_empty_rows').click(() => {
```

Comment:
why the sudden change in name convention?

Reply:
I'm following the current style of buttons in this page, i.e. `#button_enroll`

Reply:
OK